### PR TITLE
expose additional configuration options

### PR DESCRIPTION
**Pull request by arnetheduck**
* caches
* table options
* spelling fixes

### CODE REVIEW

File: rocksdb/options/cache.nim [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  ../lib/librocksdb
+
+type
+  CachePtr* = ptr rocksdb_cache_t
+
+  CacheRef* = ref object
+    cPtr*: CachePtr
+
+proc cacheCreateLRU*(size: int): CacheRef =
+  # TODO do you release it?
```
Comment (bhartnett) [inline rocksdb/options/cache.nim]: It looks like you might need to call rocksdb_cache_destroy to release the handle.

**Comment by bhartnett** [comment]
Something to be aware of is that older versions of RocksDb won't have the definitions of some of the functions if they were added later in a more recent version. When I made my changes to this wrapper library I tried to only use c functions that were defined for the version we use in development and CI. It appears that the windows CI is failing for this reason, because it can't find the definition for `rocksdb_options_set_max_background_jobs` which I'm guessing was added in a more recent version of RocksDb.

**Review by bhartnett — APPROVED** [review]

**Comment by arnetheduck** [comment]
>  When I made my changes to this wrapper library I tried to only use c functions that were defined for the version we use in development and CI. 

indeed - so there are two ways we can go from here: 

* generate the nim abi from an old header file - we usually use ubuntu lts as benchmark - this would mean pinning the vendor version that's used for static building at that version - this will avoid the guessing game of what's available at the "latest supported version"
* fully switch to the vendor solution and not look back

I think the latter makes more sense in our case, which means that we would add a windows dll build to the build process and then invert the static linking flag so it's called `rocksdbUseSystemLibs` and that the explicit option requires a flag.

File: rocksdb/options/cache.nim (hunk 1 above)
Comment (arnetheduck) [reply rocksdb/options/cache.nim]: thanks, somehow I missed it ;)

**Comment by bhartnett** [comment]
> > When I made my changes to this wrapper library I tried to only use c functions that were defined for the version we use in development and CI.
> 
> indeed - so there are two ways we can go from here:
> 
> * generate the nim abi from an old header file - we usually use ubuntu lts as benchmark - this would mean pinning the vendor version that's used for static building at that version - this will avoid the guessing game of what's available at the "latest supported version"
> * fully switch to the vendor solution and not look back
> 
> I think the latter makes more sense in our case, which means that we would add a windows dll build to the build process and then invert the static linking flag so it's called `rocksdbUseSystemLibs` and that the explicit option requires a flag.

Yes I agree. Better to make static linking the default for Linux and Mac and we can use dlls to support windows.

**Pull request merged**